crates/router/src/connector/coinbase/transformers.rs

```diff
@@ -32,9 +65,9 @@ pub struct CoinbasePaymentsRequest {
     pub cancel_url: String,
 }
 
-impl TryFrom<&types::PaymentsAuthorizeRouterData> for CoinbasePaymentsRequest {
+impl TryFrom<CoinbseRouterData<&types::PaymentsAuthorizeRouterData> for CoinbasePaymentsRequest {
     type Error = error_stack::Report<errors::ConnectorError>;
-    fn try_from(item: &types::PaymentsAuthorizeRouterData) -> Result<Self, Self::Error> {
+    fn try_from(item:&CoinbseRouterData<&types::PaymentsAuthorizeRouterData) -> Result<Self, Self::Error> {
         get_crypto_specific_payment_data(item)
```

Comment:
`item` variable is of type `&CoinbseRouterData<&types::PaymentsAuthorizeRouterData`
and `fn get_crypto_specific_payment_data` accepts the argument of type `&types::PaymentsAuthorizeRouterData` which is a mismatch.

Change the type of the argument that the function `get_crypto_specific_payment_data` is accepting and then subsequently pass the updated item in `fn get_local_price`, which would again have this mismatch.

In the function `get_local_price` use `item.amount`

You can run `cargo c` on your system to help you navigate through the type checking